### PR TITLE
Edit Profile

### PR DESCRIPTION
**Pull request by davodesign84**

### CODE REVIEW

File: packages/kauri-web/components/containers/PublicProfile/EditableHeader.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import TertiaryButton from '../../../../kauri-components/components/Button/TertiaryButton';
+import TriggerImageUploader from '../../common/ImageUploader';
+import userIdTrim from '../../../lib/userid-trim';
+
+const HeaderContainer = styled.div`
+    background-color: #1e2428;
+    display: flex;
+    align-items: flex-start;
+    color: white;
+    padding: 2.5em calc((100vw - 1280px) / 2);
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/EditableHeader.js]: Theme

File: packages/kauri-web/components/containers/PublicProfile/EditableHeader.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+ title: string,
+ avatar: string,
+ website: string,
+ twitter: string,
+ github: string,
+ name: string,
+};
+
+type Props = {
+    username: string,
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/EditableHeader.js]: Please install prettier in vscode

File: packages/kauri-web/components/containers/PublicProfile/EditableHeader.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    render() {
+        const { userId: string  } = this.props;
+        const { username, title, avatar, website, name, twitter, github} = this.state;
+        return(
+            <HeaderContainer>
+                <StyledUpload bg={avatar} handleClick={() => this.uploadImage()} text="Profile" color="white" />
+                <InputsContainers>
+                    <StyledInput onChange={(e: { target: {value: string }}) => this.setState({ name: e.target.value})} fontWeight="normal" fontSize={6} value={name} placeHolder="Add your full name" />
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/EditableHeader.js]: Extract out to this.handleChange

File: packages/kauri-web/components/containers/PublicProfile/Header.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+`;
+
+const RightSide = styled.div`
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: center;
+    flex-wrap: wrap;
+
+    h3, span {
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/Header.js]: Ew jk

File: packages/kauri-web/components/containers/PublicProfile/Header.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    }
+`;
+
+const StyledButton = styled(PrimaryButton)`
+    margin: ${props => props.theme.space[3]}px;
+    align-self: center;
+`;
+
+
+const ProfileHeader = ({ id, avatar, title, username, name, website, github, twitter, currentUser, collections, articles, toggleEditing }, props) =><PublicProfileHeader>
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/Header.js]: Was gonna say use Flowtype's inferred type '*' but it's deprecated 🤕 
https://github.com/facebook/flow/commit/39968bcd1f5badccfd4303b2245a88203fa49377

File: packages/kauri-web/components/containers/PublicProfile/Module.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { Observable } from 'rxjs/Observable';
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/Module.js]: Dude business logic without Flowtype annotations?!

File: packages/kauri-web/components/containers/PublicProfile/Module.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  });
+
+  export const saveUserDetailsEpic = (
+    action$,
+    _,
+    { apolloClient, smartContracts, web3, apolloSubscriber }
+  ) =>
+    action$
+      .ofType("SAVE_USER_DETAILS")
+      .switchMap(({ payload: { username, title, avatar, website, name, twitter, github } }) =>
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/Module.js]: Yeah use .do(() => apolloClient.resetStore())

Comment (rej156) [reply packages/kauri-web/components/containers/PublicProfile/Module.js]: Need to add analytics too

File: packages/kauri-web/components/containers/PublicProfile/View.js [hunk 8]
```diff
@@ -1,142 +1,76 @@
...
-    display: flex;
-    align-items: center;
-    justify-content: center;
-    border-radius: 50px;
-    font-size: ${props => props.theme.fontSizes[5]}px;
-    font-weight: 700;
-
-    @media (max-width: 700px) {
-        display: none;
+class PublicProfile extends Component {
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/View.js]: Has state, annotate with Flow

File: packages/kauri-web/components/containers/PublicProfile/View.js [hunk 9]
```diff
@@ -1,142 +1,76 @@
...
-            </ContentContainer>,
-        ]}
-    />}
-</div>;
+    render() {
+        const { UserQuery, ArticlesQuery, CollectionQuery, DraftsQuery, routeChangeAction, userId} = this.props;
+        if (!UserQuery.getUser || !ArticlesQuery.searchArticles || !CollectionQuery.searchCollections || !DraftsQuery.searchArticles) return null;
+        return (
+            <div>
+                {console.log(UserQuery)}
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/View.js]: console.log

File: packages/kauri-web/components/containers/PublicProfile/Header.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import styled from 'styled-components';
+import { PrimaryButton } from '../../../../kauri-components/components/Button';
+import StatisticsContainer from '../../../../kauri-components/components/PublicProfile/StatisticsContainer.bs';
+import idTrim from '../../../lib/userid-trim';
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/Header.js]: userIdTrim

**Comment by rej156** [comment]
lgtm!

File: packages/kauri-web/components/containers/PublicProfile/Header.js (hunk 5 above)
Comment (rej156) [reply packages/kauri-web/components/containers/PublicProfile/Header.js]: No inferred type in Flowtype even though it's written in OCaml 👎 :(

File: packages/kauri-web/components/containers/PublicProfile/EditableHeader.js (hunk 3 above)
Comment (rej156) [reply packages/kauri-web/components/containers/PublicProfile/EditableHeader.js]: Learn Formik, it's the new craze

File: packages/kauri-web/components/containers/PublicProfile/Header.js (hunk 10 above)
Comment (davodesign84) [reply packages/kauri-web/components/containers/PublicProfile/Header.js]: Ah that needs to be removed anyway.

File: packages/kauri-web/components/containers/PublicProfile/EditableHeader.js [hunk 11]
```diff
@@ -63,19 +63,25 @@ class EditableHeader extends Component<HeaderProps, HeaderState> {
         TriggerImageUploader(() => {}, '', (file, url: string) => this.setState({ avatar: url}));
     }
 
+    handleChange(param: string, value: string) {
```
Comment (rej156) [inline packages/kauri-web/components/containers/PublicProfile/EditableHeader.js]: Could have also done handleChange = (param: string) => (e) => {...}

**Pull request merged**